code/src/pages/products/ProductsPage.test.tsx

```diff
@@ -20,24 +23,135 @@ const mockProducts = [
 
 const mockData = { content: mockProducts, totalPages: 2, totalElements: 8 };
 
-const renderProductsPage = () =>
-  render(
-    <MemoryRouter>
-      <ProductsPage />
-    </MemoryRouter>
-  );
+const defaultQueryArguments = {
+  size: 8,
+  page: 0,
+  sort: ""
+};
+
+const testQueryArguments = (args: Partial<PaginationParams> = {}) => {
+  expect(useGetProductsQuery).toHaveBeenCalledWith({
+    ...defaultQueryArguments,
+    ...args
+  });
+};
+
+const renderAndMock = (
+  {
+    entries = "",
+    mockResponse = {}
+  }: {
+    entries?: string;
+    mockResponse?: Partial<RTKQueryReturnState<Partial<typeof mockData>>>;

```

Comment:
this type `Partial<RTKQueryReturnState<typeof mockData>>` also used for `BestSellers` tests. Can use it once

Reply:
Thanks, I created a reusable type for that inside of common types